spatialprofilingtoolbox/workflow/common/proximity.py

```diff
@@ -1,41 +1,25 @@
-"""
-Low-level calculations of proximity metric.
-"""
+"""Low-level calculations of proximity metric."""
+
 from math import isnan
 import re
 
-import numpy as np
+from pandas import DataFrame
+from sklearn.neighbors import BallTree
 
+from spatialprofilingtoolbox.db.exchange_data_formats.metrics import PhenotypeCriteria
+from spatialprofilingtoolbox.workflow.common.cell_df_indexer import get_mask
 from spatialprofilingtoolbox.standalone_utilities.log_formats import colorized_logger
 
 logger = colorized_logger(__name__)
 
 
-def get_value_and_multiindex(signature):
-    value = (1,) * len(signature['positive']) + (0,) * len(signature['negative'])
-    if len(value) == 1:
-        value = value[0]
-    multiindex = [*signature['positive'], *signature['negative']]
-    return value, multiindex
-
-def get_mask(cells, signature):
-    value, multiindex = get_value_and_multiindex(signature)
-    try:
-        loc = cells.set_index(multiindex).index.get_loc(value)
-    except KeyError:
-        return np.asarray([False,] * cells.shape[0])
-    if isinstance(loc, np.ndarray):
-        return loc
-    if isinstance(loc, slice):
-        range1 = [False,]*(loc.start - 0)
-        range2 = [True,]*(loc.stop - loc.start)
-        range3 = [False,]*(cells.shape[0] - loc.stop)
-        return np.asarray(range1 + range2 + range3)
-    if isinstance(loc, int):
-        return np.asarray([i == loc for i in range(cells.shape[0])])
-    raise ValueError(f'Could not select by index: {multiindex}. Got: {loc}')
-
-def compute_proximity_metric_for_signature_pair(signature1, signature2, radius, cells, tree):
+def compute_proximity_metric_for_signature_pair(
```

Comment:
This change broke the function's signature, there is another usage in proximity/core.py. I'm updating.

Reply:
Note that it is a bit against the design intention (as remarked elsewhere) to start using the `exchange_data_formats` pydantic types for other internal uses, i.e. for uses other than exchange via the API. Not sure what to recommend as alternative right now, so this is OK.

Reply:
Why wouldn't we want to use the exchange data formats wherever possible? The controlled types make it easier for us to understand and pass complete information as well.

Reply:
Clearly controlled types make it easier for us to understand and pass complete information.
I did not recommend uncontrolled types.
I am recommending against using the types, which were designed/selected/documented specifically for the web-application-supporting API, in the internal operation of the library. In my opinion, doing so would be an example of the anti-pattern of needlessly increasing coupling between unrelated modules.

Reply:
The kind of things I'm worried about is attributes of the exchange data types module having surprise effects on internal functions, like the bug at the top of this thread that spawned this conversation.